### PR TITLE
Suppress warnings for tf

### PR DESCRIPTION
**Pull request by shreyajain17**
Some verification left. 

- [x] A line still seems to appear from tensorflow.
- [x] Python 2.7
- [x] Python 3.6

### CODE REVIEW

File: src/python/turicreate/toolkits/_tf_utils.py [hunk 1]
```diff
@@ -14,8 +14,9 @@ def get_gpu_names():
 	"""
 	Gets the available GPU names.
 	"""
-	import tensorflow as tf
-	gpu_names = tf.config.experimental.list_physical_devices('GPU')
+	import tensorflow.compat.v1 as _tf
+	_tf.disable_v2_behavior()
```
Comment (TobyRoseman) [inline src/python/turicreate/toolkits/_tf_utils.py]: If we're writing new code, we should be using the new API.

Does the following work?
```
import TensorFlow as tf
tf.config.experimental.list_physical_devices
```

File: src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py [hunk 2]
```diff
@@ -21,6 +21,8 @@ def __init__(self, input_h, input_w, batch_size, output_size, init_weights, conf
         _tf.reset_default_graph()
 
         # Suppresses verbosity to only errors
+        import os
```
Comment (srikris) [inline src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py]: Can we make this into a utility and call it everywhere.

File: src/python/turicreate/toolkits/_tf_utils.py (hunk 1 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/_tf_utils.py]: Yeah the new one works. Yeah that does make sense which is why I had that earlier. I will revert back this change.

File: src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py (hunk 2 above)
Comment (shreyajain17) [reply src/python/turicreate/toolkits/object_detector/_tf_model_architecture.py]: Yeah that makes sense.

**Comment by srikris** [comment]
Fixes #2631

**Pull request merged**